File: CS50X/week_6/sentimental-readability/readability.py
from cs50 import get_string

def main():
    text = get_string("Text: ")

    letters = 0
    words = 1  # Start at 1 assuming there's at least one word
    sentences = 0

    for char in text:
        # Count letters (A-Z, a-z)
        if char.isalpha():
            letters += 1
        # Count words (sequences separated by spaces)
        elif char.isspace():
            words += 1
        # Count sentences (ending in . ! or ?)
        elif char in [".", "!", "?"]:
            sentences += 1

    # Calculate L and S per 100 words
    L = (letters / words) * 100
    S = (sentences / words) * 100

    # Compute index and round
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # Output results based on grade constraints
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")

if __name__ == "__main__":
    main()

File: CS50X/week_2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int grade(string t);

int main(void)
{
    string texto = get_string("Text: ");
    if (grade(texto)>16)
    {
        printf("Grade 16+\n");
    }
    else if (grade(texto)<1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n",grade(texto));
    }
}

int grade(string t)
{
    float letras = 0;
    float palavras = 1;
    float sent = 0;

    int i = 0;
    while (t[i] != '\0')
    {
        if (t[i] != '.' && t[i] != '!' && t[i] != '?' && t[i] != ',' && t[i] != ' ' && t[i] != '\'')
        {
            letras++;
        }
        if (t[i] == ' ')
        {
            palavras++;
        }
        if (t[i] == '.' || t[i] == '!' || t[i] == '?')
        {
            sent++;
        }

        i++;
    }

    float l = (letras/palavras)*100;
    float s = (sent/palavras)*100;
    float grade = 0.0588 * l - 0.296 * s - 15.8;
    ///
    //printf("letras: %f\n",letras);
    //printf("palavras: %f\n",palavras);
    //printf("sent: %f\n",sent);
    //printf("l: %f\n",l);
    //printf("s: %f\n",s);
    //printf("grade: %f\n",grade);
    ///
    return round(grade);
}
